### ISSUE TITLE
Save each test's verbose log to a file

### ISSUE THREAD

**Issue by ItalyPaleAle** [gotestyourself/gotestsum#251]
Please forgive me if this is already possible, but I couldn't find it in the docs.

When using format `standard-verbose` and tests running in parallel, it seems that the logs from all tests are shown at the same time, so it's hard to follow them.

Could gotestsum offer an option to save each test's verbose log to a separate file? So the logs can be inspected later? (This could be in addition to printing them in stdout)

**Comment by dnephin** [comment]
Thank you for opening this issue! That problem exists in other places of `gotestsum` as well. Any output from background goroutines could be attributed to any other test running at the same time. That's a limitation of `go tool test2json`.

I think to address the problem it needs to be handled in the test functions. They have access to `t.Name()` and will always attribute output to the right test.

One way I've worked around this problem is to configure the application or function being tested to send all the log output to a `bytes.Buffer`, and use a cleanup function like this to output it on failure:

```go
// *bytes.Buffer is an io.Writer, pass it to the whatever is printing
// output in place of os.Stdout
func logOutput(t *testing.T) *bytes.Buffer {
    buf := new(bytes.Buffer)
    t.Cleanup(func() {
        if t.Failed() {
            t.Log("log output", buf.String())
        }
    })
}
```

You could do the same thing with a file:

```go
// *os.File is an io.Writer, pass it to the whatever is printing
// output in place of os.Stdout
func logOutput(t *testing.T) *os.File {
    filename := strings.ReplaceAll(t.Name(), "/", "_")
    fh, err := os.Create(filename)
    if err != nil { t.Fatalf(err.Error()) }
    t.Cleanup(func() { fh.Close() })
    return fh
}
```
